docs/pages/guides/pages/signin.mdx

```diff
@@ -107,7 +108,20 @@ export default async function SignInPage() {
         <form
           action={async () => {
             "use server"
-            await signIn(provider.id)
+            try {
+              await signIn(provider.id)
+            } catch (error) {
+              // Signin can fail for a number of reasons, such as the user
+              // not existing, or the user not having the correct role.
+              // In some cases, you may want to redirect to a custom error
+              if (error instanceof AuthError) {
+                return redirect(`${SIGNIN_ERROR_URL}?error=${error.type}`)
+              }
+
+              // Otherwise if a redirects happens NextJS can handle it
+              // so you can just re-thrown the error and let NextJS handle it.
+              throw error
+            }
```

Comment:
One point of confusion to me that have work not much with NextJS was to find out that in next redirects throw an error. And `signIn` can fail. Also next auth can fail for legic reasons and is the user the responsible of managing those exceptions in their code

I tried to make this clear in the docs:
<img width="907" alt="image" src="https://github.com/nextauthjs/next-auth/assets/49499/eddccec9-88b2-47ef-b0c1-ed5aa94d55dd">

Reply:
Could you link a next.js doc here about redirects throwing?

Reply:
Done